src/main/java/edu/harvard/iq/dataverse/engine/command/impl/CuratePublishedDatasetVersionCommand.java

```diff
@@ -155,45 +164,72 @@ public Dataset execute(CommandContext ctxt) throws CommandException {
         // Update modification time on the published version and the dataset
         updateVersion.setLastUpdateTime(getTimestamp());
         tempDataset.setModificationTime(getTimestamp());
-        ctxt.em().merge(updateVersion);
-        Dataset savedDataset = ctxt.em().merge(tempDataset);
-
-        // Flush before calling DeleteDatasetVersion which calls
-        // PrivateUrlServiceBean.getPrivateUrlFromDatasetId() that will query the DB and
-        // fail if our changes aren't there
-        ctxt.em().flush();
+        newVersion = ctxt.em().merge(newVersion);
+        savedDataset = ctxt.em().merge(tempDataset);
 
         // Now delete draft version
-        DeleteDatasetVersionCommand cmd;
 
-        cmd = new DeleteDatasetVersionCommand(getRequest(), savedDataset);
-        ctxt.engine().submit(cmd);
-        // Running the command above reindexes the dataset, so we don't need to do it
-        // again in here.
+        ctxt.em().remove(newVersion);
+
+        Iterator<DatasetVersion> dvIt = savedDataset.getVersions().iterator();
+        while (dvIt.hasNext()) {
+            DatasetVersion dv = dvIt.next();
+            if (dv.isDraft()) {
+                dvIt.remove();
+                break; // We've removed the draft version, no need to continue iterating
+            }
+        }
+
+        savedDataset = ctxt.em().merge(savedDataset);
+        ctxt.em().flush();
+
+        RoleAssignment ra = ctxt.privateUrl().getPrivateUrlRoleAssignmentFromDataset(savedDataset);
+        if (ra != null) {
+            ctxt.roles().revoke(ra);
+        }
 
         // And update metadata at PID provider
-        ctxt.engine().submit(
-                new UpdateDvObjectPIDMetadataCommand(savedDataset, getRequest()));
-        
-        //And the exported metadata files
         try {
-            ExportService instance = ExportService.getInstance();
-            instance.exportAllFormats(getDataset());
-        } catch (ExportException ex) {
-            // Just like with indexing, a failure to export is not a fatal condition.
-            logger.log(Level.WARNING, "Curate Published DatasetVersion: exception while exporting metadata files:{0}", ex.getMessage());
+            ctxt.engine().submit(
+                    new UpdateDvObjectPIDMetadataCommand(savedDataset, getRequest()));
+        } catch (CommandException ex) {
+            // Make this non-fatal? This can be corrected by running the update PID API
+            // later, but who will look in the log?
```

Comment:
If we were willing to make it non-fatal, this command could be moved to `onSuccess()` as well. 
I understand the rationale of "who will look in the log? [and therefore nobody will ever get to update the stale PID metadata]". I'd say this should be a function of how confident you are that with these changes rolling back the transaction will never be a problem again. If there's any trace of doubt, a stale DataCite record is way better than lost metadata. 

The PR looks great overall.

Reply:
I guess it's rare enough, and these Curate changes are supposed to be minor enough, that catching any exception and logging is OK/maybe better than having the overall update fail.

I'll update the comment text though - it is non-fatal now, but it could cause the Curate to not succeed if the try/catch is removed.

Reply:
Doh, sorry.